### PR TITLE
Increase PHPStan to level 8

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this can only be applied to 4.x.

## Changelog

```markdown
### Changed
- `PageAdmin` now injects its required dependencies through `__construct` instead of set methods.
```

### CODE REVIEW

File: src/Admin/Extension/CreateSnapshotAdminExtension.php [hunk 1]
```diff
@@ -50,6 +50,10 @@ private function createByPage(object $object): void
     {
         if ($object instanceof PageBlockInterface) {
```
Comment (VincentLanglet) [inline src/Admin/Extension/CreateSnapshotAdminExtension.php:51]:
```
if ($object instanceof PageBlockInterface) {
    $object = $object->getPage();
}

if (!$object instanceof PageInterface) {
    return;
}

Comment (jordisala1991) [reply src/Admin/Extension/CreateSnapshotAdminExtension.php:51]: I don't like a lot to replace variables that comes from parameters, but I will try to find a cleaner option.

Comment (jordisala1991) [reply src/Admin/Extension/CreateSnapshotAdminExtension.php:51]: I didn't found a cleaner solution, I applied your patch.

**Comment by jordisala1991** [comment]
This one is probably a bit harder to review, had to change some logics around to avoid null variables @VincentLanglet .

Not sure if I followed the best option in each specific case.

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**